File: src/Courses.java
public class Courses {

    private String name;
    private String courseCode;
    private String prerequisite;
    private int credit;
    private int courseType;
    private int semester;

    public Courses(String name, String courseCode, String prerequisite, int credit, int courseType, int semester, int theoreticalCourseHour, int practicalLessonHour) {
        this.name = name;
        this.courseCode = courseCode;
        this.prerequisite = prerequisite;
        this.credit = credit;
        this.courseType = courseType;
        this.semester = semester;
        this.theoreticalCourseHour = theoreticalCourseHour;
        this.practicalLessonHour = practicalLessonHour;
    }

    private int theoreticalCourseHour;
    private int practicalLessonHour;

    public void setCourseType(int courseType) {
        this.courseType = courseType;
    }

    public void setSemester(int semester) {
        this.semester = semester;
    }

    public void setTheoreticalCourseHour(int theoreticalCourseHour) {
        this.theoreticalCourseHour = theoreticalCourseHour;
    }

    public void setPracticalLessonHour(int practicalLessonHour) {
        this.practicalLessonHour = practicalLessonHour;
    }

    public int getSemester() {
        return semester;
    }

    public int getTheoreticalCourseHour() {
        return theoreticalCourseHour;
    }

    public int getPracticalLessonHour() {
        return practicalLessonHour;
    }

    public int getCourseType() {
        return courseType;
    }

    public String getName() {
        return name;
    }

    public String getCourseCode() {
        return courseCode;
    }

    public String getPrerequisite() {
        return prerequisite;
    }

    public int getCredit() {
        return credit;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }

    public void setPrerequisite(String prerequisite) {
        this.prerequisite = prerequisite;
    }

    public void setCredit(int credit) {
        this.credit = credit;
    }

    public Courses(){







    }

}

File: courses.py
class Courses:
    def __init__(self, data):
        self.__list_of_students = None
        self.__quota = None
        self.__name = data['name']
        self.__course_code = data['courseCode']
        self.__prerequisite = data['prerequisite']
        self.__credit = data['credit']
        self.__course_type = data['courseType']
        self.__semester = data['semester']
        self.__course_year = data['courseYear']
        self.__theoretical_course_hour = data['theoreticalCourseHour']
        self.__practical_lesson_hour = data['practicalLessonHour']

    def set_quota(self, quota):
        self.__quota = quota

    def set_course_type(self, courseType):
        self.__course_type = courseType

    def set_semester(self, semester):
        self.__semester = semester

    def set_theoretical_course_hour(self, theoretical_course_hour):
        self.__theoretical_course_hour = theoretical_course_hour

    def set_practical_lesson_hour(self, practical_lesson_hour):
        self.__practical_lesson_hour = practical_lesson_hour

    def get_course_type(self):
        return self.__course_type

    def get_name(self):
        return self.__name

    def get_course_code(self):
        return self.__course_code

    def get_prerequisite(self):
        return self.__prerequisite

    def get_credit(self):
        return self.__credit

    def set_name(self, name):
        self.__name = name

    def set_course_code(self, course_code):
        self.__course_code = course_code

    def set_course_year(self, course_year):
        self.__course_year = course_year

    def set_prerequisite(self, prerequisite):
        self.__prerequisite = prerequisite

    def set_credit(self, credit):
        self.__credit = credit

    def get_semester(self):
        return self.__semester

    def get_course_year(self):
        return self.__course_year

    def get_theoretical_course_hour(self):
        return self.__theoretical_course_hour

    def get_practical_lesson_hour(self):
        return self.__practical_lesson_hour

    def check_if_prerequisite(self, courses):
        return courses.get_prerequisite[0] != ""

    def check_if_two_prerequisite(self, courses):
        if courses.get_prerequisite()[0] != "" and courses.get_prerequisite()[1] != "":
            return True
        else:
            return False

    def get_prerequisite_name(self):
        return self.__prerequisite[0]

    def get_two_prerequisite_name(self, courses):
        preRequisiteNames = [courses.get_prerequisite()[0], courses.get_prerequisite()[1]]
        return preRequisiteNames

    def add_to_list_of_students(self, student):
        self.__list_of_students.append(student)
